src/views/RegistrationView.vue

```diff
@@ -0,0 +1,108 @@
+<script setup>
+import { ref } from 'vue'
+import { useStore } from 'vuex'
+import { useRouter } from 'vue-router'
+
+const store = useStore()
+const router = useRouter()
+const profileBaseUrl = store.state.userProfileUrl
+
+const username = ref('')
+const password = ref('')
+const errorMessage = ref('')
+const showPassword = ref(false)
+
+// validation rule for vuetify components
+// we can add rules here in the future for passwords
+const rules = {
+    required: value => !!value || 'Required.'
+}
+
+async function authenticateAndGetToken() {
+    const authUrl = profileBaseUrl + 'api-token-auth/'
+    const credentials = {
+      username: username.value,
+      password: password.value
+  }
+
+  try {
+    const response = await fetch(authUrl, {
```

Comment:
Should we make a function in the api.js file for these requests to keep it consistent? 
with this modification the function can take an optional token parameter. 

Did not check how the OCS frontend handles it as it uses a passthrough component, maybe Matt or Jon have another api util file example with a different solution for handling multiple apis with and without tokens.

```
async function fetchApiCall({ url, method, body = null, successCallback = null, failCallback = null, token=null }) {
  const headers = {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
// this line needs some changing to only include the key Authorization if theres a token
    'Authorization': token ? token : 'Token 123456789abcdefg', 
  }
```

Reply:
You could also add an optional headers param, so it's more flexible than just a token!

Reply:
Matt's idea is more customizable so I agree